lib/jorb.ex

```diff
@@ -6,35 +6,117 @@ defmodule Jorb do
+  Jorb is a simple queue-based jobs processing system for Elixir.
+  Works great with Amazon SQS.
 
   ## How
 
-  Modules use `Jorb.Job` and implement its' `c:queue_name/0` and `c:perform/1` callbacks
-
-  Example:
+  Define your job module:
 
   ```
-  defmodule Demo.Jobs.TestJob do
-     use Jorb.Job
+  defmodule HelloWorld.Job do
+    use Jorb.Job
+
+    def read_queues do
+      ["high_priority_greetings", "regular_greetings"]
+    end
+
+    def write_queue(greeting) do
+      if greeting["name"] == "Zeke" do
+        "high_priority_greetings"
+      else
+        "regular_greetings"
+      end
+    end
 
-     def queue_name, do: "test"
-     def perform(name) do
-       IO.puts("Hello #{name}")
-     end
+    def perform(greeting) do
+      IO.puts "Hello, #{greeting["name"]}!"
+      :ok
+    end
   end
   ```
 
-  Then, queue jobs to be performed later with `perform_async`
+  Enqueue work:
+
+  ```
+  HelloWorld.Job.enqueue(%{"name" => "Ray"})
+  ```
+
+  Perform work:
+
+  ```
+  # poll queues once
+  HelloWorld.Job.work(read_timeout: 1000, perform_timeout: 5000)
+
+  # poll queues forever
+  HelloWorld.Job.workers(worker_count: 2, read_interval: 1000)
+  |> Supervisor.start_link(strategy: :one_for_one)
+  ```
+
+  ## Installation
 
-  ```Demo.Jobs.TestJob.perform_async("Andy")```
+  Put the following into your `mix.exs` file's `deps` function:
 
-  And sometime later "Hello Andy" will be output to the console
+      {:jorb, "~> 0.3.0"}
 
+  ## Configuration
+
+  In order of priority, configs can be provided by:
+
+  * Passing options in the `opts` parameter to each function
+  * Configuring your job module in `config/config.exs`:
+
+      config :jorb, HelloWorld.Job, [read_timeout: 5000]
+
+  * Configuring global Jorb settings in `config/config.exs`:
+
+      config :jorb, write_batch_size: 10
+
+  Options:
+
+  * `:backend` - the module implementing `Jorb.Backend`, default
+    `Jorb.Backend.Memory`. You should set this to something
+    else (like `Jorb.Backend.SQS` in production.
+  * `:worker_count` - number of workers to launch per job module,
+    default `System.schedulers_online()`.
+  * `:write_batch_size` - number of messages to write at once, default 1.
+  * `:write_interval` - milliseconds to wait before flushing outgoing
+     messages, default 1000.
+  * `:read_batch_size` - number of messages to read at once, default 1.
+  * `:read_interval` - milliseconds to sleep between fetching messages,
+     default 1000.
+  * `:read_duration` - milliseconds to hold connection open when polling
+    for messages, default 1000.
+  * `:read_timeout` - milliseconds before giving up when reading messages,
+    default 2000.
+  * `:perform_timeout` - milliseconds before giving up when performing a
+    single job, default 5000.
 
   """
 
-  def backend() do
-    Application.get_env(:jorb, :backend)
+  @defaults [
+    backend: Jorb.Backend.Memory,
+    write_interval: 1000,
+    write_batch_size: 1,
```

Comment:
It appears that we're not doing anything with this param. Should this get added or is that for a separate enhancement?

Reply:
I'm doing the Jorb queuer next, this would hook into that.